src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -2302,17 +2327,150 @@ DynamicMetaObject IDynamicMetaObjectProvider.GetMetaObject(Expression parameter)
+            get => _flags.HasFlag(PSObjectFlags.IsDeserialized);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.IsDeserialized;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.IsDeserialized;
+                }
+            }
+        }
+
+        private bool StoreTypeNameAndInstanceMembersLocally
+        {
+            get => _flags.HasFlag(PSObjectFlags.StoreTypeNameAndInstanceMembersLocally);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.StoreTypeNameAndInstanceMembersLocally;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.StoreTypeNameAndInstanceMembersLocally;
+                }
+            }
+        }
 
         internal bool IsHelpObject
         {
-            get => _isHelpObject;
-            set => _isHelpObject = value;
+            get => _flags.HasFlag(PSObjectFlags.IsHelpObject);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.IsHelpObject;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.IsHelpObject;
+                }
+            }
         }
 
-        private bool _isHelpObject = false;
+        internal bool HasGeneratedReservedMembers
+        {
+            get => _flags.HasFlag(PSObjectFlags.HasGeneratedReservedMembers);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.HasGeneratedReservedMembers;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.HasGeneratedReservedMembers;
+                }
+            }
+        }
 
-        #endregion
+        internal bool ImmediateBaseObjectIsEmpty
+        {
+            get => _flags.HasFlag(PSObjectFlags.ImmediateBaseObjectIsEmpty);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.ImmediateBaseObjectIsEmpty;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.ImmediateBaseObjectIsEmpty;
+                }
+            }
+        }
+
+
+        internal WriteStreamType WriteStream
+        {
+            get => _writeStream;
+            set => _writeStream = value;
+        }
+
+        /// <summary>
+        /// Gets an instance member if it's name matches the predicate. Otherwise null.
+        /// </summary>
+        internal PSPropertyInfo GetFirstPropertyOrDefault(MemberNamePredicate predicate)
+        {
+            if (_instanceMembers != null)
+            {
+                foreach (var instanceMember in _instanceMembers)
+                {
+                    if (instanceMember is PSPropertyInfo propInfo)
```

Comment:
```suggestion
                    if (instanceMember is PSPropertyInfo propInfo && predicate.Invoke(propInfo.Name))
```

These nested `if`s seem to be mostly unneeded here, I think; collapsing them looks a bit clearer to me.